framework/src/node/generator/endpoint.ts

```diff
@@ -0,0 +1,229 @@
+	pool: TransactionPool;
+	broadcaster: Broadcaster;
+}
+
+interface EndpointInit {
+	logger: Logger;
+	generatorDB: KVStore;
+	blockchainDB: KVStore;
+}
+
+interface EndpointContext {
+	params: Record<string, unknown>;
+	logger: Logger;
+}
+
+export class Endpoint {
+	private readonly _keypairs: dataStructures.BufferMap<Keypair>;
+	private readonly _generators: Generator[];
+	private readonly _consensus: Consensus;
+	private readonly _stateMachine: StateMachine;
+	private readonly _pool: TransactionPool;
+	private readonly _broadcaster: Broadcaster;
+	private readonly _chain: Chain;
+
+	private _logger!: Logger;
+	private _generatorDB!: KVStore;
+	private _blockchainDB!: KVStore;
+
+	public constructor(args: EndpointArgs) {
+		this._keypairs = args.keypair;
+		this._generators = args.generators;
+		this._consensus = args.consensus;
+		this._chain = args.chain;
+		this._stateMachine = args.stateMachine;
+		this._pool = args.pool;
+		this._broadcaster = args.broadcaster;
+	}
+
+	public init(args: EndpointInit) {
+		this._logger = args.logger;
+		this._generatorDB = args.generatorDB;
+		this._blockchainDB = args.blockchainDB;
+	}
+
+	public async postTransaction(ctx: EndpointContext): Promise<PostTransactionResponse> {
+		const reqErrors = validator.validate(postTransactionRequestSchema, ctx.params);
+		if (reqErrors?.length) {
+			throw new LiskValidationError(reqErrors);
+		}
+		const req = (ctx.params as unknown) as PostTransactionRequest;
+		const transaction = this._chain.dataAccess.decodeTransaction(
+			Buffer.from(req.transaction, 'hex'),
+		);
+		const txContext = new TransactionContext({
+			eventQueue: new EventQueue(),
+			logger: this._logger,
+			networkIdentifier: this._chain.constants.networkIdentifier,
+			stateStore: new StateStore(this._blockchainDB),
+			transaction,
+		});
+		const result = await this._stateMachine.verifyTransaction(txContext);
+		if (result.status === VerifyStatus.FAIL) {
+			throw new InvalidTransactionError(
+				result.error?.message ?? 'Transaction verification failed.',
+				transaction.id,
+			);
+		}
+		if (this._pool.contains(transaction.id)) {
+			return {
+				transactionId: transaction.id.toString('hex'),
+			};
+		}
+
+		// Broadcast transaction to network if not present in pool
+		this._broadcaster.enqueueTransactionId(transaction.id);
+
+		const { error } = await this._pool.add(transaction);
+		if (error) {
+			this._logger.error({ err: error }, 'Failed to add transaction to pool.');
+			throw new InvalidTransactionError(
+				error.message ?? 'Transaction verification failed.',
+				transaction.id,
+			);
+		}
+
+		this._logger.info(
+			{
+				id: transaction.id,
+				nonce: transaction.nonce.toString(),
+				senderPublicKey: transaction.senderPublicKey,
+			},
+			'Added transaction to pool',
+		);
+		return {
+			transactionId: transaction.id.toString('hex'),
+		};
+	}
+
+	public async updateForgingStatus(ctx: EndpointContext): Promise<UpdateForgingStatusResponse> {
```

Comment:
This function is copied from `forger.ts` and updated to use `EndpointContext` as input.
Also, BFT properties checks are handled in `_consensus.verifyGeneratorInfo` and `_consensus.isSynced` functions instead